### PR TITLE
CHET-530: Fix startup and prerequisite check errors

### PR DESCRIPTION
**Pull request by tarka**

### CODE REVIEW

File: core/src/main/kotlin/com/atlassian/migration/datacenter/core/db/DefaultDatabaseExtractorFactory.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * limitations under the License.
+ */
+package com.atlassian.migration.datacenter.core.db
+
+import com.atlassian.migration.datacenter.core.application.ApplicationConfiguration
+import com.atlassian.migration.datacenter.core.application.DatabaseConfiguration.DBType
+import com.atlassian.migration.datacenter.spi.exceptions.DatabaseMigrationFailure
+
+class DefaultDatabaseExtractorFactory(val config: ApplicationConfiguration) : DatabaseExtractorFactory {
+    override val extractor: DatabaseExtractor by lazy {
```
Comment (theghostwhoforks) [inline core/src/main/kotlin/com/atlassian/migration/datacenter/core/db/DefaultDatabaseExtractorFactory.kt:23]: This is nice!

File: core/src/main/java/com/atlassian/migration/datacenter/core/application/DatabaseConfigurationXmlElement.java [hunk 2]
```diff
@@ -60,44 +63,38 @@ private void validateRequiredValues(String... values) throws ConfigurationReadEx
...
-            throw new ConfigurationReadException("Database configuration file has invalid values");
+            throw new ConfigurationReadException("Database configuration file has invalid or missing values");
         }
     }
 
-    public boolean isDataSourcePresent() {
-        return dbConfigXmlElement != null;
-    }
-}
+    public Optional<URI> getURI() {
```
Comment (theghostwhoforks) [inline core/src/main/java/com/atlassian/migration/datacenter/core/application/DatabaseConfigurationXmlElement.java:70]: Does this need to be public?

File: core/src/main/java/com/atlassian/migration/datacenter/core/application/DatabaseConfigurationXmlElement.java [hunk 3]
```diff
@@ -60,44 +63,38 @@ private void validateRequiredValues(String... values) throws ConfigurationReadEx
...
     }
 
-    public String getUrl() {
-        return url;
-    }
-
-    public String getUserName() {
-        return userName;
-    }
+    public Optional<DatabaseConfiguration.DBType> getDBType() {
```
Comment (theghostwhoforks) [inline core/src/main/java/com/atlassian/migration/datacenter/core/application/DatabaseConfigurationXmlElement.java:85]: same comment here - does this need to be public?

File: core/src/main/java/com/atlassian/migration/datacenter/core/application/DatabaseConfigurationXmlElement.java (hunk 2 above)
Comment (tarka) [reply core/src/main/java/com/atlassian/migration/datacenter/core/application/DatabaseConfigurationXmlElement.java:70]: Not necessarily, but there is no harm in making them so, and it opens them up to use later.

**Pull request merged**